pkg/udev/udev_test.go

```diff
@@ -0,0 +1,47 @@
+// +build linux,cgo
+
+/*
+Copyright 2018 OpenEBS Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package udev_test
+
+import (
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/openebs/node-disk-manager/pkg/udev"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewDeviceFromSysPath(t *testing.T) {
+	diskType := "disk"
+	disks, err := exec.Command("lsblk", "-r", "-d", "-n", "-o", "type,name,maj:min").Output()
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, device := range strings.Split(string(disks[:len(disks)-1]), "\n") {
+		parts := strings.Split(device, " ")
+		if parts[0] != "disk" {
+			continue
+		}
+		syspath := "/sys/dev/block/" + parts[2]
+		device := udev.NewDeviceFromSysPath(syspath)
+		assert.Equal(t, diskType, device.Devtype())
```

Comment:
The intent is to make sure the NewDeviceFromSysPath returns a valid structure for a valid syspath. The test should include checking with - valid and invalid syspath.

Reply:
`NewDeviceFromSysPath()` returns a pointer of `udev_device` to validate this struct we can validate device struct with corresponding property of disk as value of that struct is empty.

Reply:
add test case for invalid syspath